### PR TITLE
Feature: Update zkdb sdk to latest

### PR DESCRIPTION
**Pull request by SangTran-127**
## Type

- [x] Feature
- [ ] Bug fix
- [ ] CI/CD
- [ ] Documentation
- [ ] Other

## Description

Brief description of the changes made.

### CODE REVIEW

File: packages/smart-contract/src/network/api.ts [hunk 1]
```diff
@@ -46,13 +44,13 @@ export class BlockberryApi {
         options
       );
       if (!response.ok) {
-        logger.error('response:', response);
+        console.error('response:', response);
```
Comment (chiro-hiro) [inline packages/smart-contract/src/network/api.ts:47]: Why are you change this to `console.error`?

File: packages/smart-contract/src/network/native-api.ts [hunk 2]
```diff
@@ -42,11 +44,11 @@ export async function sendTransaction(json: any, url: string): Promise<TSendTran
     const { data, errors } = response.data;
 
     if (errors) {
-      throw Error(errors)
+      throw errors;
     }
 
-    return data;
+    return data.sendZkapp.zkapp;
```
Comment (chiro-hiro) [inline packages/smart-contract/src/network/native-api.ts:50]: Is this any better way to validate before return?

File: packages/smart-contract/src/network/native-api.ts [hunk 3]
```diff
@@ -42,11 +44,11 @@ export async function sendTransaction(json: any, url: string): Promise<TSendTran
...
   } catch (error) {
-    throw Error(error as any)
+    throw Error(error as any);
```
Comment (chiro-hiro) [inline packages/smart-contract/src/network/native-api.ts:52]: Is this try catch necessary? you literally just catch and throw it.

File: packages/zkdb/src/sdk/signer/nodejs-signer.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ export class NodeSigner implements Signer {
 
   constructor(privateKey: PrivateKey, endpoint: string) {
     this.privateKey = privateKey;
-    this.client = new Client({ network: 'mainnet' });
+    this.client = new Client({ network: 'testnet' });
```
Comment (chiro-hiro) [inline packages/zkdb/src/sdk/signer/nodejs-signer.ts:16]: Can we stop hardcode this? please make it a configurable instead.

File: packages/smart-contract/src/network/api.ts (hunk 1 above)
Comment (SangTran-127) [reply packages/smart-contract/src/network/api.ts:47]: If we want to use logger from framework, smart-contract package now require `@orochi-network/framework` and the user that use zkdb need to install it.

File: packages/smart-contract/src/network/native-api.ts (hunk 2 above)
Comment (SangTran-127) [reply packages/smart-contract/src/network/native-api.ts]: Yep, I added the validate on my new commit

File: packages/smart-contract/src/network/native-api.ts (hunk 3 above)
Comment (SangTran-127) [reply packages/smart-contract/src/network/native-api.ts]: I think it redundant, removed

File: packages/zkdb/src/sdk/signer/nodejs-signer.ts (hunk 4 above)
Comment (SangTran-127) [reply packages/zkdb/src/sdk/signer/nodejs-signer.ts]: Fixed, I changed to passing network, these endpoint and networkId base on user now.

```ts
const signer = new NodeSigner(privateKey, 'testnet')
```

File: packages/zkdb/src/sdk/signer/nodejs-signer.ts [hunk 5]
```diff
@@ -1,20 +1,26 @@
...
 
-  constructor(privateKey: PrivateKey, endpoint: string) {
+  private network: NetworkId
+  constructor(privateKey: PrivateKey, network: NetworkId) {
     this.privateKey = privateKey;
+    this.network = network
+    if (network === 'testnet') {
+      this.client = new Client({network: 'testnet'})
+      this.endpoint = 'https://api.minascan.io/node/devnet/v1/graphql'
+    }
```
Comment (chiro-hiro) [inline packages/zkdb/src/sdk/signer/nodejs-signer.ts:21]: Missing else, network will be overwritten

File: packages/smart-contract/src/network/api.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import logger from "src/helper/logger";
+import { logger } from '../utils';
 
 export type TChain = 'mainnet' | 'devnet';
```
Comment (chiro-hiro) [inline packages/smart-contract/src/network/api.ts:3]: We only need one network enum instead of have multiple type.

**Review by chiro-hiro — APPROVED** [review]

**Pull request merged**